src/main/java/com/example/vaccinecalendar/service/ChildService.java

```diff
@@ -24,5 +24,8 @@ public void addChild(ChildModel child){
     public List<ChildModel> getChildList(){
         return childRepository.findAll();
     }
+    public ChildModel getChildById(Long id){
+        return childRepository.findById(id).orElse(null);
```

Comment:
To rozwiązanie ze zwracaniem nulli mi się nie podoba.

Kontroler powinien raczej zwracać 404 jeśli ID nie zostanie znalezione. Jeśli chcesz w bardziej ambitny sposób rozwiązywać taką sytuację to można wprowadzić customowe redirecty do wyspecjalizowanych stron z błędami, albo przez @ControllerAdvice albo ręcznie.